compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertLoopsForVectorization.cpp

```diff
@@ -66,12 +66,29 @@ class AMDAIEInsertLoopsForVectorizationPass
   // Return success if the generic op is rewritten, failure otherwise.
   LogicalResult maybeRewrite(linalg::GenericOp genericOp,
                              IRRewriter &rewriter) {
+    if (isa<linalg::CopyOp, linalg::FillOp>(genericOp)) return failure();
+
     auto iteratorTypes = genericOp.getIteratorTypesArray();
     auto numIterators = iteratorTypes.size();
 
     // No outer dimensions to tile if fewer than 4 iterators.
     if (numIterators < 4) return failure();
 
+    // Enable generating loops for vectorization in case of element-wise ops.
```

Comment:
Can you add a comment motivating the decision to tile all but the final 2 dimensions? 

Is this because the final 2 dimensions correspond to the m and n dimensions which are correspondingly tiled in this pass for matmul?